src/model/Authenticator/AuthenticatorRepository.ts

```diff
@@ -18,7 +18,7 @@ export class AuthenticatorRepository extends BaseRepository<AuthenticatorDTO> {
       (doc: { id: string }) => [doc.id],
     );
     const result = await query({ id: userId });
-    return result.toArray();
+    return Array.isArray(result) ? result : result.toArray();
```

Comment:
Tests shouldn't alter the implementation. Try using `mockResolveValueOnce()` in the mock to return a valid item for this particular test.